src/components/Dashboard.jsx

```diff
@@ -1,7 +1,206 @@
-import React from "react";
+import React, { useContext, useEffect, useState } from "react";
+import { UserContext } from "../context/UserContext";
+import { useRef } from "react";
 
 const Dashboard = () => {
-  return <div className="container">Dashboard</div>;
+  const [postList, setPostList] = useState([]);
+  const textareaRef = useRef();
+  const [postTitle, setPostTitle] = useState("");
+  const { name } = useContext(UserContext);
+
+  useEffect(() => {
+    let retrievedPosts;
+    if (localStorage.getItem("posts")) {
+      retrievedPosts = JSON.parse(localStorage.getItem("posts"));
+      setPostList(retrievedPosts);
+    }
+  }, []);
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    const newPost = {
+      id: new Date().getTime(),
+      title: postTitle,
+      date: new Date(),
+      author: name,
+      isPostOpen: false,
+      likeCount: 0,
+      isLiked: false,
+    };
+
+    if (postTitle) {
+      postList.push(newPost);
+      localStorage.setItem("posts", JSON.stringify(postList));
+    }
+
+    setPostTitle("");
+    textareaRef.current.focus();
+  };
+
+  const handleDelete = (id) => {
+    const deletedItem = postList.filter((item) => item.id !== id);
+    setPostList(deletedItem);
+    localStorage.setItem("posts", JSON.stringify(deletedItem));
+  };
+
+  const openCommentBox = (id) => {
+    const newPosts = [...postList];
+    const curPost = postList.find((post) => post.id === id);
+    curPost.isPostOpen = !curPost.isPostOpen;
+    setPostList(newPosts);
+  };
+
+  const handleLikeCount = (id) => {
+    const newPosts = [...postList];
+    const curPost = postList.find((post) => post.id === id);
+
+    if (curPost.isLiked) {
+      curPost.isLiked = false;
+      curPost.likeCount = curPost.likeCount - 1;
+    } else {
+      curPost.isLiked = true;
+      curPost.likeCount = curPost.likeCount + 1;
+    }
+
+    setPostList(newPosts);
+    localStorage.setItem("posts", JSON.stringify(newPosts));
+  };
```

Comment:
without taking the currentPost,  can't we destructure the newPosts and update it? I think that's the best practice

Reply:
I tried it but it says something like `assigning values to constants`. Need your assistance.

Reply:
Bhai we can do something like this in our function

`setPostList((prevPostList) => {
  return prevPostList.map((post) => {
    return post.id === id ? { ...post, isPostOpen: !post.isPostOpen } : post;
  });
});`